docs/user/adding-glean-to-your-project.md

```diff
@@ -270,6 +270,19 @@ Please refer to the [custom pings documentation](pings/custom.md).
 
 All of Glean's target languages use a separate worker thread to do most of Glean's work, including any I/O. This thread is fully managed by Glean as an implementation detail. Therefore, users should be free to use the Glean API wherever it is most convenient, without worrying about the performance impact of updating metrics and sending pings.
 
+{{#include ../tab_header.md}}
+
+<div data-lang="Python" class="tab">
+When using the Python bindings, Glean's work is done on a separate thread, managed by Glean itself.
+Glean releases the Global Interpreter Lock (GIL), therefore your application's threads should not be in contention with Glean's thread.
+
+Glean installs an [`atexit` handler](https://docs.python.org/3/library/atexit.html) so the Glean thread can attempt to cleanly shut down when your application exits.
+This handler will wait up to 1 second for any pending work to complete.
+If that times out, some Glean work may be lost.
```

Comment:
@mdboom , on other platforms I/O and networking happen on a thread that's separate from the "dispatcher" queue for API calls.

If we did something similar here, thus being consistent with Kotlin/Swift, we'd mitigate data loss: API/ping serialization are _very likely_ to finish < 1s, networking operations are more likely to take more time on weird network conditions.

Reply:
Yep -- I have this bug for the follow-up: https://bugzilla.mozilla.org/show_bug.cgi?id=1626403

In the case of Python, I think we'll use a separate process (rather than a thread) so that networking can continue even if the main process finishes, but it's the same concept.  My only hesitation on the follow-up is whether to do it before or after the Rust networking refactor lands.

Reply:
Will the work actually be lost? Or would Glean be smart enough to resend it the next time the command / application is invoked?

Reply:
Oh, yeah, of course.  If networking is off, say, the ping file won't be deleted and a reattempt will be made next time.  So pretty low risk.

Reply:
I'd suggest mentioning that, since the current wording is a little scary.

Reply:
In general, right now, I think there's a _minor_ chance of loosing a small fraction of data: since I/O and networking are happening on the same dispatcher, if any work is queued _after_ this lengthy operations **and** the timeout is hit, the work queued after is lost, I think.

If B is a lengthy task on which we hit the 1s timeout, then the following can happen:

Queued tasks: A-B (timeout) - C (lost)- D (lost) - E (lost)